arviz/tests/test_plots.py

```diff
@@ -387,6 +401,18 @@ def test_plot_pair_bad(models, model_fit):
         plot_pair(obj, var_names=["mu"])
 
 
+@pytest.mark.parametrize("has_stats", [True, False])
+def test_plot_pair_divergences_warning(has_stats):
+    data = load_arviz_data("centered_eight")
+    if has_stats:
+        data.sample_stats = data.sample_stats.rename({"diverging": "diverging_missing"})

```

Comment:
So if `has_stats` is true that means we set `diverging` to `diverging_mission`?

Should this be the opposite case semantically?

Reply:
The warning can be due to 2 reasons, either the InferenceData object has no `sample_stats` group, or the `sample_stats` dataset has no `diverging` variable. In the test, `has_stats=True` indicates the second case, whereas `has_stats=False` is the first case, the input only has a posterior group. 
Maybe I should modify the message of the warning as it is basically thought for the second case.